test/sql-luatest/view_test.lua

```diff
@@ -0,0 +1,437 @@
+        t.assert_equals(res, "test")
+
+        -- Can't create view with incorrect SELECT statement.
+        box.space.test:drop()
+        -- This case must fail since parser converts it to expr AST.
+        raw_sp[6].sql = 'SELECT 1;'
+        exp_err = "Failed to execute SQL statement: SELECT 1;"
+        t.assert_error_msg_equals(exp_err, box.space._space.replace,
+                                  box.space._space, raw_sp)
+
+        box.execute("DROP VIEW v1;")
+        box.execute("DROP TABLE t1;")
+
+        -- Can't drop space via Lua if at least one view refers to it.
+        box.execute('CREATE TABLE t1(id INT PRIMARY KEY);')
+        t.assert(box.space.t1 ~= nil)
+        box.execute('CREATE VIEW v1 AS SELECT * FROM t1;')
+        t.assert(box.space.v1 ~= nil)
+        exp_err = "Can't drop space 't1': other views depend on this space"
+        t.assert_error_msg_equals(exp_err, box.space.t1.drop, box.space.t1)
+        box.execute('DROP VIEW v1;')
+        box.execute('DROP TABLE t1;')
+
+        -- Check that alter transfers reference counter.
+        box.execute("CREATE TABLE t1(id INTEGER PRIMARY KEY);")
+        t.assert(box.space.t1 ~= nil)
+        box.execute("CREATE VIEW v1 AS SELECT * FROM t1;")
+        t.assert(box.space.v1 ~= nil)
+        _, err = box.execute("DROP TABLE t1;")
+        exp_err = "Can't drop space 't1': other views depend on this space"
+        t.assert_equals(err.message, exp_err)
+        sp = box.space._space:get{box.space.t1.id}
+        sp = box.space._space:replace(sp)
+        _, err = box.execute("DROP TABLE t1;")
+        exp_err = "Can't drop space 't1': other views depend on this space"
+        t.assert_equals(err.message, exp_err)
+        box.execute("DROP VIEW v1;")
+        box.execute("DROP TABLE t1;")
+
+        -- Try to create invalid view using direct insert to space _space.
+        box.execute("CREATE TABLE b (s1 INT PRIMARY KEY);")
+        t.assert(box.space.b ~= nil)
+        box.execute("CREATE TABLE c (s1 INT PRIMARY KEY);")
+        t.assert(box.space.c ~= nil)
+        box.execute([[CREATE VIEW bcv(x, y) AS VALUES((SELECT 'k' FROM b),
+                     (VALUES((SELECT 1 FROM b WHERE s1 IN
+                     (VALUES((SELECT 1 + c.s1 FROM c)))))))]])
+        t.assert(box.space.bcv ~= nil)
+        local space_tuple = box.space._space.index[0]:max():totable()
+        space_tuple[1] = space_tuple[1] + 1 -- id
+        space_tuple[3] = space_tuple[3] .. '1' -- name
+        space_tuple[6].sql = string.gsub(space_tuple[6].sql, 'FROM c',
+                                                             'FROM ccc')
+        exp_err = "Space 'ccc' does not exist"
+        t.assert_error_msg_equals(exp_err, box.space._space.insert,
+                                  box.space._space, space_tuple)
+        box.space.bcv:drop()
+        box.execute("DROP TABLE c;")
+        box.execute("DROP TABLE b;")
+
+        --
+        -- Make sure we can't alter a view.
```

Comment:
In this test you use two comment styles:
```Lua
        -- Try to create invalid view using direct insert to space _space.
```
and
```Lua
        --
        -- Make sure we can't alter a view.
        --
```

Let's use the first style inside a test and the second style outside a test (i.e. inside a test we have `--` only before filled comment lines, and outside a test we will start and end with lines containing only `--`).